### PR TITLE
Reorganise components

### PR DESCRIPTION
**Pull request by malcolmbaig**
## Context
Reviewing how this directory is organised as it grows in size.

## Changes proposed in this pull request

- Move interface-specific components into the appropriate directory.
- Move all components that are used in more than one interface into `app/components/shared`.

The remaining components in the root directory are generic parts of the UI that are used throughout the app.

## Guidance to review

- General observation - common pairings of shared components are provider/support and api_docs/support.
- There doesn't seem to be much of an organisational or complexity improvement from duplicating the shared components in the specific namespaces.
- Does the shared directory make sense? 
- Is it worth moving the generic components in the root directory into a folder as well? Might discourage accumulating components at the root, and prompt a bit more thought about how components are organised.
- When reviewing, it's probably more useful to look at the structure of the directory on this branch rather than the diff.

## Link to Trello card
https://trello.com/c/HwBPqDv1

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

**Comment by TobyRet** [comment]
I think this is definitely a step in the right direction. Perhaps one potential point of confusion with the new structure is that 'generic' components at the root would according to the definition above deemed to be 'sharable'? 

Possible options:

- having a 'generic' or 'lib' folder for generic components
- renaming the 'shared' folder to 'core' or 'app'

Naming stuff is hard

**Comment by benswannack** [comment]
I agree with @TobyRet , it would be nice to put the 'utility' like components somewhere rather than just the app/components root.

**Comment by TobyRet** [comment]
> I agree with @TobyRet , it would be nice to put the 'utility' like components somewhere rather than just the app/components root.

@malcolmbaig @ben-swa (if it helps) I went through an old React project I worked on and the 'utility' like components were kept in a separate 'layout' folder

**Comment by malcolmbaig** [comment]
@ben-swa @TobyRet I think both 'utility' and 'layout' are good options. Going with 'utility' for now, as layout is a slightly overloaded term in a Rails context.

**Review by TobyRet — APPROVED** [review]
LGMT @malcolmbaig - you have a linter error otherwise good to merge

File: app/components/utility/filter_component.html.erb [hunk 1]
```diff
@@ -26,7 +26,7 @@
               <ul class="moj-filter-tags">
                 <% tags_for_active_filter(active_filter).each do |tag| %>
                   <li>
-                    <%= link_to(tag[:remove_link], class: 'app-checkbox-filter__tag', id: "tag-#{tag[:title].parameterize}") do %>
+                    <%= govuk_link_to(tag[:remove_link], class: 'app-checkbox-filter__tag', id: "tag-#{tag[:title].parameterize}") do %>
```
Comment (benswannack) [inline app/components/utility/filter_component.html.erb:29]: This file was previously ignored in the config. If we're fixing the ignored error, can we remove the relevant ignore lines? Not sure why we explicitly ignored it though – maybe it's supposed to be a non-govuk link?

Comment (malcolmbaig) [reply app/components/utility/filter_component.html.erb]: Good spot—have just updated/cleaned up the ignores for now.

**Review by benswannack — APPROVED** [review]
Nice 🚢

**Pull request merged**